coderd/httpmw/logger_internal_test.go

```diff
@@ -0,0 +1,154 @@
+package httpmw
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/tracing"
+	"github.com/coder/coder/v2/testutil"
+	"github.com/coder/websocket"
+)
+
+func TestRequestLogger_WriteLog(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	sink := &fakeSink{}
+	logger := slog.Make(sink)
+	logger = logger.Leveled(slog.LevelDebug)
+	logCtx := NewRequestLogger(logger, "GET", time.Now())
+
+	// Add custom fields
+	logCtx.WithFields(
+		slog.F("custom_field", "custom_value"),
+	)
+
+	// Write log for 200 status
+	logCtx.WriteLog(ctx, http.StatusOK)
+
+	if len(sink.entries) != 1 {
+		t.Fatalf("expected 1 log entry, got %d", len(sink.entries))
+	}
+
+	if sink.entries[0].Message != "GET" {
+		t.Errorf("expected log message to be 'GET', got '%s'", sink.entries[0].Message)
+	}
```

Comment:
Use the `require` package, it has a lot of neat utilities for assertions and automatically writes nice output on test failure: https://pkg.go.dev/github.com/stretchr/testify/require

(note: there's an `assert` package and a `require` package in the same module, they are practically the same EXCEPT `require` will immediately fail the test and `assert` will let it continue but fail at the end)

e.g.
```suggestion
	require.Len(t, sink.entries, 1)
	require.Equal(t, "GET", sink.entries[0].Message)
```

Reply:
Definitely cleaner. Will add.